### PR TITLE
Airport Challenge Pull Request

### PR DESCRIPTION
**Pull request by eripheebs**

### CODE REVIEW

**Comment by mechacoach** [comment]
You had your airport_challenge reviewed by **Marcus Bullock**

### The good points are:

* Overview of project in the README
* `context` or `describe` blocks are used (where appropriate) to group tests
* The solution avoids vacuous tests
* `before` blocks are used where appropriate
* There are a minimal number of untested paths in the code
* Unit tests all have single expectations
* Random behaviour is stubbed where appropriate
* The code follows Ruby naming conventions, Readability is supported by consistent naming
* Guard clauses are used where appropriate, Booleans are returned implicitly where appropriate, removeallcommented-outcode: There is no commented-out code in the submission
* Code avoids unnecessary public methods, Code avoids public access to internal objects (e.g. planes array)
* Classes have single responsibilities, Each methods has a single responsibility
* Code avoids strings where it should use symbols
* Command/Query naming is not an issue in the code
* The solution does not have any redundant code

### You should consider the following improvements:

* Installation Instructions should be in the README
* Use named subject with `described_class`
* Get rid of redundant `respond_to` expectations
* Remove all commented-out code
* Avoid magic numbers (e.g. on capacity)
* Improper use of attr_*

### Additional comments:
Very good., avoidmagicnumberse.g.oncapacity: Code avoids magic numbers

see https://github.com/makersacademy/airport_challenge/blob/master/docs/review.md for more details

File: lib/airport.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  private
+
+    attr_reader :planes
+
+    def in_this_airport?(plane)
+      @planes.include? plane
+    end
+
+    def is_stormy?
+      Weather.new.stormy?
```
Comment (danldb) [inline lib/airport.rb]: you could initialise the airport like so:
```ruby
def initialize(weather: Weather.new)
  @weather = Weather
end
```

File: lib/plane.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+class Plane
+
+  def initialize
+    @landed = false
+  end
+
+  def is_landed
```
Comment (danldb) [inline lib/plane.rb]: method naming - use a 'commanding' verb - you are instructing the plane to land. `plane.land` would be appropriate.

File: spec/airport_spec.rb [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      expect{subject.takeoff(plane)}.to raise_error "That plane is not currently in this airport."
+    end
+
+    it 'expects airport to have a capacity that is set to the default capacity if nothing is specified' do
+      expect(subject.capacity).to eq described_class::DEFAULT_CAPACITY
+    end
+
+    it 'expects the capacity to be able to be set when initialized' do
+      airport = Airport.new 13
+      expect(airport.capacity).to eq 13
```
Comment (danldb) [inline spec/airport_spec.rb]: why are you testing the capacity directly? give it a capacity of 1 and then try and land 2 planes. Capacity should be private.

File: spec/plane_spec.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      expect(subject.landed?).to eq(false)
+    end
+
+  end
+
+  describe 'landed?' do
+
+    it 'expects it to be able to confirm if a plane is landed' do
+      is_expected.to respond_to(:landed?)
+    end
```
Comment (danldb) [inline spec/plane_spec.rb]: you could write:
```ruby
it { is_expected.to respond_to(:landed?) }
```

File: spec/plane_spec.rb [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+      expect(subject.landed?).to eq true
+    end
+
+  end
+
+  describe 'took_off' do
+
+        it 'expects it to be able to change the value of landed to true' do
+          subject.took_off
+          expect(subject.landed?).to eq false
```
Comment (danldb) [inline spec/plane_spec.rb]: at this point you are testing the landed message, so the previous respond_to test is now redundant - you should delete it to save subsequent dev a few lines of reading.

**Pull request closed without merging**